packages/sanity/src/core/form/studio/assetSource/DefaultSource.tsx

```diff
@@ -24,8 +29,9 @@ const buildQuery = (start = 0, end = PER_PAGE, assetType = ASSET_TYPE_IMAGE) =>
     mimeType,
     extension,
     size,
-    metadata {dimensions}
-  }
+    metadata {dimensions},
+    "acceptedType": ${acceptParam.length <= 0 ? 'true' : `mimeType == "${acceptParam}"`}
```

Comment:
when the type is "any" then the file and image input send in '' strings which means that the type will be accepted